cost/aws/savings_plan/recommendations/aws_savings_plan_recommendations.pt

```diff
@@ -279,32 +279,40 @@ script "js_sp_cleanup", type: "javascript" do
-          currentMaximumHourlyOnDemandSpend: (Math.round(reco["currentMaximumHourlyOnDemandSpend"] * 100) / 100).toString(10),
-          currentMinimumHourlyOnDemandSpend: (Math.round(reco["currentMinimumHourlyOnDemandSpend"] * 100) / 100).toString(10),
-          estimatedAverageUtilization: (Math.round(reco["estimatedAverageUtilization"] * 100) / 100).toString(10),
-          estimatedMonthlySavingsAmount: (Math.round(reco["estimatedMonthlySavingsAmount"] * 100) / 100 * currency_adjustment).toString(10),
-          estimatedOnDemandCost: (Math.round(reco["estimatedOnDemandCost"] * 100) / 100).toString(10),
-          estimatedROI: (Math.round(reco["estimatedROI"] * 100) / 100).toString(10),
-          estimatedSavingsAmount: (Math.round(reco["estimatedSavingsAmount"] * 100) / 100).toString(10),
-          estimatedSavingsPercentage: (Math.round(reco["estimatedSavingsPercentage"] * 100) / 100).toString(10),
-          estimatedSavingsPlanCost: (Math.round(reco["estimatedSavingsPlanCost"] * 100) / 100).toString(10),
-          hourlyCommitmentToPurchase: (Math.round(reco["hourlyCommitmentToPurchase"] * 100) / 100).toString(10),
-          instanceFamily: reco["instanceFamily"],
-          offeringId: reco["offeringId"],
-          region: reco["region"],
-          upfrontCost: (Math.round(reco["upfrontCost"] * 100) / 100).toString(10),
-          estimatedMonthlySavingsAmountWithCurrencyCode : (Math.round(reco["estimatedMonthlySavingsAmount"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          estimatedSavingsPlanCostWithCurrencyCode : (Math.round(reco["estimatedSavingsPlanCost"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          hourlyCommitmentToPurchaseWithCurrencyCode : (Math.round(reco["hourlyCommitmentToPurchase"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          upfrontCostWithCurrencyCode : (Math.round(reco["upfrontCost"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          currentAverageHourlyOnDemandSpendWithCurrencyCode : (Math.round(reco["currentAverageHourlyOnDemandSpend"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          currentMaximumHourlyOnDemandSpendWithCurrencyCode : (Math.round(reco["currentMaximumHourlyOnDemandSpend"] * 100) / 100).toString(10)+" "+reco["currencyCode"],
-          currentMinimumHourlyOnDemandSpendWithCurrencyCode : (Math.round(reco["currentMinimumHourlyOnDemandSpend"] * 100) / 100).toString(10)+" "+reco["currencyCode"]
+      var resourceType = param_savings_plan_type
+      if (reco["instanceFamily"] != "" && reco["instanceFamily"] != null) {
+        resourceType += ", " + reco["instanceFamily"]
+      }
+      result.push({
+        accountID: reco["accountID"],
+        accountName: accountName,
+        currencyCode: reco["currencyCode"],
+        currentAverageHourlyOnDemandSpend: (Math.round(reco["currentAverageHourlyOnDemandSpend"] * 100) / 100).toString(10),
```

Comment:
Looks like
```
Math.round(reco["currentAverageHourlyOnDemandSpend"] * 100) / 100)
```
is quite redundant since after multiplying by 100 you divide by 100, that multiplication gets cancelled by that division.

![image](https://github.com/flexera-public/policy_templates/assets/54189123/762fd15c-4d31-43b0-849a-c23066e3bb34)

I don't know, despite being redundant it could be probably necessary, it's like the coconut of the source code of Team Fortress 2: https://www.thegamer.com/this-coconut-jpg-in-team-fortress-2s-game-files-if-deleted-breaks-the-game-and-no-one-knows-why/

Reply:
In javascript we use that method to round decimal numbers. It is redundant if we look at it logically, but internally, what it does is that it rounds to the number of zeros we add: https://stackoverflow.com/questions/11832914/how-to-round-to-at-most-2-decimal-places-if-necessary

Reply:
That makes sense, thank you!!!

Reply:
This is fine, but just leaving a note to let you know this could be done in an even easier way using toFixed():

> (Math.round(reco["currentAverageHourlyOnDemandSpend"] * 100) / 100).toString(10)
'1234.12'
> reco["currentAverageHourlyOnDemandSpend"].toFixed(2)
'1234.12'